.lgtm.yml

```diff
@@ -13,6 +13,12 @@ extraction:
         requirements:
           - cython>=0.29
   cpp:
+    prepare:
+      packages:
+        - python3-pip
+    configure:
+      command:
+        - pip3 install cython
     index:
       build_command:
         - python3 setup.py build
```

Comment:
Did you see something that indicates this is necessary? LGTM is failing sproadically, I doubt it is due to missing cython since the build command succeeds, it is during import something goes wrong

Reply:
Here's part of an a build log from LGTM C/C++ before I made the change to .lgtm.yml:

```
[2019-06-17 04:49:59] [build] + python3 setup.py build
[2019-06-17 04:50:05] [build] deptrace-server: File '/usr/bin/cython' found in package cython 
[2019-06-17 04:50:05] [build] deptrace-server: Running installation program: /opt/deptrace/sudo-install 
[2019-06-17 04:50:06] [build] Reading package lists... 
[2019-06-17 04:50:06] [build] Building dependency tree... 
[2019-06-17 04:50:06] [build] Reading state information... 
[2019-06-17 04:50:07] [build] Suggested packages: 
[2019-06-17 04:50:07] [build] cython-doc 
[2019-06-17 04:50:07] [build] The following NEW packages will be installed: 
[2019-06-17 04:50:07] [build] cython 
[2019-06-17 04:50:07] [build] 0 upgraded, 1 newly installed, 0 to remove and 94 not upgraded. 
[2019-06-17 04:50:07] [build] Need to get 2,102 kB of archives. 
[2019-06-17 04:50:07] [build] After this operation, 9,609 kB of additional disk space will be used. 
[2019-06-17 04:50:07] [build] Get:1 http://archive.ubuntu.com/ubuntu cosmic/universe amd64 cython amd64 0.28.4-1 [2,102 kB] 
[2019-06-17 04:50:08] [build] debconf: delaying package configuration, since apt-utils is not installed 
[2019-06-17 04:50:08] [build] Fetched 2,102 kB in 0s (15.6 MB/s) 
[2019-06-17 04:50:09] [build] Selecting previously unselected package cython. 
[2019-06-17 04:50:11] [build] (Reading database ... 69161 files and directories currently installed.) 
[2019-06-17 04:50:11] [build] Preparing to unpack .../cython_0.28.4-1_amd64.deb ... 
[2019-06-17 04:50:11] [build] Unpacking cython (0.28.4-1) ... 
[2019-06-17 04:50:11] [build] Setting up cython (0.28.4-1) ... 
[2019-06-17 04:50:13] [build] Processing triggers for man-db (2.8.4-2) ... 
[2019-06-17 04:50:13] [build] deptrace-server: Package installer exited with code 0
```
It installed cython 0.28.4.  What is the minimum version required to build numpy?  The file `INSTALL.rst.txt` says "Cython >= 0.29.2 (for development versions of numpy, not for released versions)"

Reply:
With the modified .lgtm.yml, cython 0.29.10 is installed:

```
[2019-06-17 07:00:03] [build] Collecting cython
[2019-06-17 07:00:04] [build] Downloading https://files.pythonhosted.org/packages/5d/7e/d2a81d821193c88113e3613f2df456a4d7b1b15bd4551e97978e8107e3ef/Cython-0.29.10-cp36-cp36m-manylinux1_x86_64.whl (2.1MB) 
[2019-06-17 07:00:04] [build] Installing collected packages: cython 
[2019-06-17 07:00:05] [build] Successfully installed cython-0.29.10
```

Reply:
but it still fails

Reply:
but it still fails

Reply:
xref the [LGTM fourm issue](https://discuss.lgtm.com/t/c-c-ananlysis-of-numpy-fails/2148), I will put a link to the cython log there

Reply:
> but it still fails

Yup.  The commit message says "experiment";  it doesn't say *successful* experiment. :smile:

> xref the LGTM fourm issue, I will put a link to the cython log there

Thanks.